solutions/ex3/Main.hs

```diff
@@ -202,6 +216,22 @@ createSortInclusionInfer ''VarL    ''LhsL
 createSortInclusionInfer ''ExpL    ''RhsL
 createSortInclusionInfer ''AssignL ''StatementL
 
+--- Question: why is it enough to just convert AssignL to StatementL as sorts?
+-- An `Assign` constructor has more data than a `Statement` constructor.
+-- A `Block` constructor has more data than a `BlockStmt` constructor.
+-- I am assuming that the Template-generated code will automatically omit unnecessary data when converting an `Assign` to a `Statement`?
+
+-- The task was described as "replace instances of the `ImpAssign` node by the generic `Assign` node".
+-- Comparing the constructors:
+
+-- ImpAssign :: e VarL -> e ExpL -> Statement e StatementL
+-- Assign :: e LhsL -> e AssignOpL -> e RhsL -> Assign e AssignL
+
+-- I expected that I will have to write code converting `Assign` to `ImpAssign` by omitting unnecessary data.
+-- But it was never necessary to write such code. Maybe I'm confused about how the generated code works?
```

Comment:
In this exercise, that's not asked. 

If you look at the various Common/Trans.hs files in Cubix, you'll see such conversions in both directions.